server/sonar-server/src/main/java/org/sonar/server/setting/ws/ListDefinitionsAction.java

```diff
@@ -113,12 +108,11 @@ private static ListDefinitionsRequest toWsRequest(Request request) {
   }
 
   private Optional<ComponentDto> getComponent(DbSession dbSession, ListDefinitionsRequest wsRequest) {
-    String componentId = wsRequest.getComponentId();
-    String componentKey = wsRequest.getComponentKey();
-    if (componentId != null || componentKey != null) {
-      return Optional.of(componentFinder.getByUuidOrKey(dbSession, componentId, componentKey, ID_AND_KEY));
+    String componentKey = wsRequest.getComponent();
```

Comment:
Good point, but we'll have a lot of issues as the Java plugin doesn't like public class to return Optional...